tensorflow/compiler/mlir/tosa/transforms/legalize_tfl.cc

```diff
@@ -2062,6 +2063,28 @@ LogicalResult ConvertTFLFillOp::matchAndRewrite(
   return success();
 }
 
+LogicalResult ConvertTFLReduceAllOp::matchAndRewrite(
+    Operation* op, PatternRewriter& rewriter) const {
+  auto tfl_all_op = cast<TFL::ReduceAllOp>(op);
+
+  RankedTensorType output_type =
+      tfl_all_op.getResult().getType().dyn_cast<RankedTensorType>();
+  if (!output_type) return failure();
+
+  ElementsAttr axes_elems;
+  if (!matchPattern(tfl_all_op.getReductionIndices(), m_Constant(&axes_elems)))
+    return failure();
+
+  llvm::Optional<Value> result = convertReduceAllOp(
+      rewriter, op, output_type, tfl_all_op.getInput(), axes_elems);
+
+  if (!result) return failure();
```

Comment:
So this would only fail if the input was not ranked and you had an unranked input but ranked output?

Reply:
Yes, I think so. I'm not sure a tensor having inconsistent input and output will pass down from higher-level. Will it happen?

It seems that similar rank checks both for input and output exist elsewhere. Do you prefer to remove this check?